Resource reconcilation leads to null pointer error with referencePath

Was trying out provide referencePath (Random value since I do not have specific info what info should go in) in resource_with_ResourceRef.yaml file ,

This lead to null pointer issue:
ocm-controller logs 
2023-03-25T05:22:55Z    LEVEL(-4)       ocm-component-version-reconcile found component {"controller": "componentversion", "controllerGroup": "delivery.ocm.software", "controllerKind": "ComponentVersion", "ComponentVersion": {"name":"vault","namespace":"ocm-system"}, "namespace": "ocm-system", "name": "vault", "reconcileID": "c1ab2ca5-f8c5-4e8e-b72b-a4f007fc0287", "component": {"apiVersion": "delivery.ocm.software/v1alpha1", "kind": "ComponentVersion", "namespace": "ocm-system", "name": "vault"}}
2023-03-25T05:22:55Z    INFO    resource-controller     found resource  {"controller": "resource", "controllerGroup": "delivery.ocm.software", "controllerKind": "Resource", "Resource": {"name":"vault-helm","namespace":"ocm-system"}, "namespace": "ocm-system", "name": "vault-helm", "reconcileID": "8d3b2551-8e20-4a93-a4c7-b5c253e1b593", "resource": {"apiVersion": "delivery.ocm.software/v1alpha1", "kind": "Resource", "namespace": "ocm-system", "name": "vault-helm"}}
2023-03-25T05:22:55Z    INFO    Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference        {"controller": "resource", "controllerGroup": "delivery.ocm.software", "controllerKind": "Resource", "Resource": {"name":"vault-helm","namespace":"ocm-system"}, "namespace": "ocm-system", "name": "vault-helm", "reconcileID": "8d3b2551-8e20-4a93-a4c7-b5c253e1b593"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1d60056]

goroutine 347 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x2311500, 0x3e050a0})
        /opt/hostedtoolcache/go/1.20.1/x64/src/runtime/panic.go:884 +0x213
github.com/open-component-model/ocm-controller/pkg/ocm.(*Client).GetResource(0xc0005d0f00, {0x2a4bf40, 0xc000c24c90}, 0xc000734200, {{0xc000497fb0, 0x10}, {0x0, 0x0}, 0x0, {0xc0005cc290, ...}})
        /home/runner/work/ocm-controller/ocm-controller/pkg/ocm/ocm.go:84 +0x1f6
github.com/open-component-model/ocm-controller/controllers.(*ResourceReconciler).reconcile(0xc0006ed450, {0x2a4bf40, 0xc000c24c90}, 0xc000734200, 0xc000c40000)
        /home/runner/work/ocm-controller/ocm-controller/controllers/resource_controller.go:211 +0x2cb
github.com/open-component-model/ocm-controller/controllers.(*ResourceReconciler).Reconcile(0xc0006ed450, {0x2a4bf40, 0xc000c24c90}, {{{0xc000497f80?, 0x0?}, {0xc000497f76?, 0x40dea7?}}})
        /home/runner/work/ocm-controller/ocm-controller/controllers/resource_controller.go:165 +0x877
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2a4bf40?, {0x2a4bf40?, 0xc000c24c90?}, {{{0xc000497f80?, 0x2228aa0?}, {0xc000497f76?, 0x0?}}})
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00027adc0, {0x2a4be98, 0xc0006ed2c0}, {0x23d9da0?, 0xc0004568a0?})
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:323 +0x35f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00027adc0, {0x2a4be98, 0xc0006ed2c0})
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:231 +0x587

git commit id of sample project 
https://github.com/mayankraipure12/helm-example-demo/tree/efc89319523bd34b6ce7d7dc359a84a9a9df9851/components

--- a/controllers/mutation_reconcile_looper.go
+++ b/controllers/mutation_reconcile_looper.go
@@ -389,6 +389,10 @@ func (m *MutationReconcileLooper) compileMapping(ctx context.Context, cv *v1alph
 		return nil, err
 	}
 
+	if cd == nil {
+		return nil, fmt.Errorf("component descriptor not found with ref: %+v", cv.Status.ComponentDescriptor.ComponentDescriptorRef)
+	}
+
 	// first create the component descriptor struct
 	root := cueCtx.CompileString("component:{}").FillPath(cue.ParsePath("component"), cueCtx.Encode(cd.Spec))
 

--- a/pkg/ocm/ocm.go
+++ b/pkg/ocm/ocm.go
@@ -80,6 +80,10 @@ func (c *Client) GetResource(ctx context.Context, cv *v1alpha1.ComponentVersion,
 		return nil, "", fmt.Errorf("failed to find component descriptor for reference: %w", err)
 	}
 
+	if cd == nil {
+		return nil, "", fmt.Errorf("component descriptor not found for reference path: %+v", resource.ReferencePath)
+	}
+
 	identity := v1alpha1.Identity{
 		v1alpha1.ComponentNameKey:    cd.Name,
 		v1alpha1.ComponentVersionKey: cd.Spec.Version,

